core/src/scopymainwindow_api.cpp

```diff
@@ -0,0 +1,147 @@
+#include "scopymainwindow_api.h"
+#include <pluginbase/scopyjs.h>
+#include "qapplication.h"
+using namespace scopy;
+
+Q_LOGGING_CATEGORY(CAT_SCOPY_API, "Scopy_API")
+
+
+ScopyMainWindow_API::ScopyMainWindow_API(ScopyMainWindow *w)
+	: ApiObject()
+	, m_w(w)
+{
+}
+
+ScopyMainWindow_API::~ScopyMainWindow_API()
+{
+}
+
+void ScopyMainWindow_API::acceptLicense()
+{
+	if (m_w->license) {
```

Comment:
Can you register the object after creating the whole ui ? so you no longer need these checks ? You could switch them to Q_ASSERT for safety.